AsyncDisplayKit/Details/ASRangeControllerBeta.mm

```diff
@@ -38,16 +40,49 @@ - (instancetype)init
   }
   
   _rangeIsValid = YES;
+  _currentRangeMode = ASLayoutRangeModeCount;
   
   return self;
 }
 
+- (void)dealloc
+{
+  if (_didRegisterForNotifications) {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+  }
+}
+
 #pragma mark - Core visible node range managment API
 
++ (ASLayoutRangeMode)rangeModeForInterfaceState:(ASInterfaceState)interfaceState
+                                scrollDirection:(ASScrollDirection)scrollDirection
+                               currentRangeMode:(ASLayoutRangeMode)currentRangeMode
+{
+  // If we used full mode, don't switch to minimum mode. That will destroy all the hard work done before.
+  if (currentRangeMode == ASLayoutRangeModeFull) {
```

Comment:
At some point, it is critical that we do release the very expensive backing stores created for the display range in particular. The fetch data range probably make sense to keep her around for a while longer, although the minimum fetch data range is still pretty generous in size. An example here is switching from feed to library. It probably makes sense to release most of the Display range in such a situation. Similarly, if a user scrolls through a pin and then swipes horizontally to another one, we should reclaim most of the display range from the previous one. I think the same is true for zooming into content, Like opening a related pin. I think the uniting theme is that if we are no longer visible, We should fall back to the minimum range?